pkg/git/poll/gitrepopolljob.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2021-2024 SUSE LLC
+package poll
+
+import (
+	"context"
+	"fmt"
+
+	v1alpha1 "github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	"github.com/reugn/go-quartz/quartz"
+	"golang.org/x/sync/semaphore"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ quartz.Job = &GitRepoPollJob{}
+
+type GitFetcher interface {
+	LatestCommit(ctx context.Context, gitrepo *v1alpha1.GitRepo, client client.Client) (string, error)
+}
+
+type GitRepoPollJob struct {
+	sem     *semaphore.Weighted
+	client  client.Client
+	GitRepo v1alpha1.GitRepo
+	fetcher GitFetcher
+}
+
+func gitRepoPollDescription(gitRepo v1alpha1.GitRepo) string {
+	return fmt.Sprintf("gitrepo-%s-%s", gitRepo.Namespace, gitRepo.Name)
+}
+
+func GitRepoPollKey(gitRepo v1alpha1.GitRepo) *quartz.JobKey {
+	return quartz.NewJobKeyWithGroup(gitRepo.Name, gitRepo.Namespace)
+}
+
+func NewGitRepoPollJob(c client.Client, f GitFetcher, gitRepo v1alpha1.GitRepo) *GitRepoPollJob {
+	return &GitRepoPollJob{
+		sem:     semaphore.NewWeighted(1),
+		client:  c,
+		GitRepo: gitRepo,
+		fetcher: f,
+	}
+}
+
+func (j *GitRepoPollJob) Execute(ctx context.Context) error {
+	if !j.sem.TryAcquire(1) {
+		// already running
+		return nil
+	}
+	defer j.sem.Release(1)
+
+	j.fetchLatestCommitAndUpdateStatus(ctx)
+
+	return nil
+}
+
+func (j *GitRepoPollJob) Description() string {
+	return gitRepoPollDescription(j.GitRepo)
+}
+
+func (j *GitRepoPollJob) fetchLatestCommitAndUpdateStatus(ctx context.Context) {
+	logger := ctrl.Log.WithName("git-latest-commit-poll-watch")
+	commit, err := j.fetcher.LatestCommit(ctx, &j.GitRepo, j.client)
+	if err != nil {
+		logger.Error(err, "error fetching commit", "gitrepo name", j.GitRepo.Name)
```

Comment:
I'd recommend using `gitrepo_name` to make this easier to query for in log storage.

Reply:
changed

Reply:
This is a left-over from following the advice in https://github.com/kubernetes-sigs/controller-runtime/blob/main/TMP-LOGGING.md#logging-structured-values-key-value-pairs

> Use lower-case, space separated keys. For example object for objects, api version for APIVersion

I think, most of the code (see `WithValues`) has been switched to camelCase.